geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/query/QueryProcess.scala

```diff
@@ -98,6 +102,10 @@ class QueryVisitor(features: SimpleFeatureCollection,
     source.getFeatures(query)
   }
 
+  override def getExpressions: java.util.List[Expression] = {
+    val (_, finalSFT) = QueryPlanner.buildTransformSFT(origSft, properties)
+    FilterHelper.propertyNames(filter, finalSFT).map(ff.property).toList
```

Comment:
Ah, guess I misunderstood, that does make sense though, finalSFT could make it retype.